core/src/main/scala/kafka/zk/AdminZkClient.scala

```diff
@@ -125,6 +125,10 @@ class AdminZkClient(zkClient: KafkaZkClient) extends Logging {
         throw new InvalidReplicaAssignmentException("Duplicate replica assignment found: " + partitionReplicaAssignment)
     )
 
+    val partitionSize = partitionReplicaAssignment.size
+    if (partitionReplicaAssignment.keys.reduce(_+_) != partitionSize * (partitionSize - 1) / 2)
```

Comment:
instead of making this partitionSize * (partitionSize - 1) / 2) part of If, lets make this calculation ahead in a val